presto-main/src/main/java/io/prestosql/server/protocol/QueryResultRows.java

```diff
@@ -280,6 +282,10 @@ public QueryResultRows build()
 
         private static Optional<List<ColumnAndType>> combine(List<Column> columns, List<Type> types)
```

Comment:
We'd need to annotate these parameters with `@Nullable` as well. Instead of pushing nulls further, let's move the null checks into the caller.